cruise-control-client/cruisecontrolclient/client/Endpoint.py

```diff
@@ -159,6 +158,27 @@ def compose_endpoint(self) -> str:
         if self.parameter_name_to_value:
             combined_parameter_to_value.update(self.parameter_name_to_value)
 
+        return combined_parameter_to_value
+
+    def compose_endpoint(self) -> str:
+        warnings.warn("This method is deprecated as of 1.0.0, as it needlessly recreates requests functionality. "
```

Comment:
Nit (there is one more similar usage): Should the deprecation warning be placed after or before documentation?

Reply:
While PEP appears to be silent on the issues, I think the current placement actually breaks proper parsing of the docstring.

Accordingly, move the `DeprecationWarning` after the docstring, where applicable.
3a16e82c57936577d2d347235f2ffba06ffe10cf